internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -0,0 +1,105 @@
+package graph
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+type countResolveCheck struct {
```

Comment:
I'd lean in favor of removing this code as it is a pseudo-mock. We already have a mocked CheckResolver.

```go
mockController := gomock.NewController(t)
defer mockController.Finish()
mockCheckResolver := NewMockedCheckResovler(mockController)
```

We can make the appropriate assertions on calls to `ResolveCheck` on the mock. For example,
```go
mockCheckResolver.EXPECT().
	ResolveCheck(gomock.Any(), gomock.Any()).
	Return(&ResolveCheckResponse{Allowed: true}, nil).
	MaxTimes(1)
```

See the example below in my other comment for a full example.

Reply:
change to MockCheckResolver